src/lib/client.ts

```diff
@@ -472,13 +477,44 @@ export class CustomClient extends Client {
     if (targetChannelId) filter.targetChannelId = targetChannelId;
     await AutoPingModel.deleteMany(filter);
   }
-
+  u;
   async getAllAutoPing(guildId: string): Promise<IAutoPing[]> {
     return await AutoPingModel.find({
       guildId: guildId,
     });
   }
 
+  async addEmoji(guildId: string, name: string): Promise<void> {
+    const millis15min = 15 * 60 * 1000;
+    const time = Date.now();
+    const emojiUsage =
+      (await EmojiUsageModel.findOne({ guildId: guildId, name: name })) ??
+      (await EmojiUsageModel.create({
+        guildId: guildId,
+        name: name,
+        count: 0,
+        lastUsage: null,
+      }));
+    if (emojiUsage.lastUsage !== null && time - emojiUsage.lastUsage < millis15min) {
+      return;
+    }
+    const delta = time - emojiUsage.lastUsage;
+
+    emojiUsage.count *= Math.pow(emojiUsageDecayPerDay, delta / millisToDay);
+    emojiUsage.count++;
+    emojiUsage.lastUsage = time;
+    await emojiUsage.save();
+  }
+
+  async listEmoji(guildId: string): Promise<EmojiUsage[]> {
+    const time = Date.now();
+    const emojis = await EmojiUsageModel.find({ guildId: guildId });
+    return emojis.filter((usage) => {
+      const decay = Math.pow(emojiUsageDecayPerDay, time - usage.lastUsage);
+      return new EmojiUsage(usage.name, usage.count * decay);
+    });
```

Comment:
What? Why are we `.filter`-ing but having the return of the filter be an object? This won't filter the output at all. Did you intend to `.map`?